testbench_waitress.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Request's HTTPChannel with socket of waitress is exposed to produce "broken-stream" and "return-reset-connection" error."""
+
+from waitress.channel import HTTPChannel
+from waitress.task import WSGITask
+from waitress import create_server
+
+
+class testbench_WSGITask(WSGITask):
+    def get_environment(self):
+        environ = super().get_environment()
+        environ["waitress.channel"] = self.channel
+        return environ
+
+
+class testbench_HTTPChannel(HTTPChannel):
+    task_class = testbench_WSGITask
+
+
+def testbench_create_server(
+    application, map=None, _start=True, _sock=None, _dispatcher=None, **kw
+):
+    if map is None:
+        map = {}
```

Comment:
Did you make any action?

Reply:
No, as when I removed default it gave error while starting testbench. 

So Basically this default value( i.e. `None`) of map is not unit tested because in unit test ( 
 `test_created_wsgi_server_should_have_testbench_httpChannel ` ) I need to initialize and provide `map` to testbench_create_server(..) so that I can assert populated wsgi server in the `map`.  
But still we need to keep  `None` as default value of map because in ` waitress\__init__.py ` server ( which is return of testbench_create_server ) is invoked only with application( ` server = _server(app, **kw) ` )

Please let me know if you want me to try something else.

Reply:
Ack.  Maybe a comment explaining that is only intended for testing?

Reply:
Or should we add `# pragma: no cover ` to exclude this line from coverage ?

Reply:
I think it is more important to say why than to ignore the coverage problem.  Your call.

Reply:
Got it. Added the comment. Please have a look, after it I'll squash the commits into one and merge.